src/librustc_typeck/check/writeback.rs

```diff
@@ -160,8 +162,9 @@ impl<'cx, 'gcx, 'tcx> Visitor<'gcx> for WritebackCx<'cx, 'gcx, 'tcx> {
 
         self.visit_node_id(e.span, e.id);
 
-        if let hir::ExprClosure(_, _, body, _) = e.node {
+        if let hir::ExprClosure(_, _, body, _, _) = e.node {
             let body = self.fcx.tcx.hir.body(body);
+            // FIXME: Why visit the args here?
```

Comment:
But `visit_body` only visits the node IDs for the arguments (as opposed to the argument patterns) with `visit_id` (rather than a nonexistent `visit_arg` or something), which `WritebackCx` can't do anything with (because it e.g. doesn't have a span).

That's probably worth a comment.